lib/apiservers/service/restapi/handlers/vch_get.go

```diff
@@ -0,0 +1,464 @@
+				PortGroup: &models.ManagedObject{
+					ID: mobidToID(value.Common.ID),
+				},
+				Nameservers: *asIPAddresses(&value.Nameservers),
+				Gateway: &models.Gateway{
+					Address:             asIPAddress(value.Gateway.IP),
+					RoutingDestinations: []models.IPRange{asIPRange(&value.Gateway)},
+				},
+				IPRanges: *asIPRanges(&value.Destinations),
+			})
+		}
+	}
+	model.Network.Container = containerNetworks
+
+	// storage
+	scratchSize := int(vchConfig.Storage.ScratchSize)
+	model.Storage = &models.VCHStorage{
+		BaseImageSize: asKB(&scratchSize),
+	}
+
+	volumeLocations := make([]*models.VCHStorageVolumeStoresItems0, 0, len(vchConfig.Storage.VolumeLocations))
+	for label, path := range vchConfig.Storage.VolumeLocations {
+		parsedPath := object.DatastorePath{}
+		parsed := parsedPath.FromString(path.Path)
+		if parsed {
+			path.Path = parsedPath.Path
+		}
+
+		volume := models.VCHStorageVolumeStoresItems0{Datastore: path.String(), Label: label}
+		volumeLocations = append(volumeLocations, &volume)
+	}
+	model.Storage.VolumeStores = volumeLocations
+
+	imageStores := make([]string, 0, len(vchConfig.Storage.ImageStores))
+	for _, value := range vchConfig.Storage.ImageStores {
+		imageStores = append(imageStores, value.String())
+	}
+	model.Storage.ImageStores = imageStores
+
+	// auth
+	model.Auth = &models.VCHAuth{
+		Client: &models.VCHAuthClient{},
+	}
+
+	if vchConfig.Certificate.HostCertificate != nil {
+		model.Auth.Server = &models.VCHAuthServer{
+			Certificate: asPemCertificate(vchConfig.Certificate.HostCertificate.Cert),
+		}
+	}
+
+	model.Auth.Client.CertificateAuthorities = asPemCertificates(vchConfig.Certificate.CertificateAuthorities)
+
+	// endpoint
+	model.Endpoint = &models.VCHEndpoint{
+		Memory: asMiB(&d.MemoryMB),
+		CPU: &models.VCHEndpointCPU{
+			Sockets: int64(d.NumCPUs),
+		},
+		OperationsCredentials: &models.VCHEndpointOperationsCredentials{
+			User: vchConfig.Connection.Username,
+			// Password intentionally excluded from GET responses for security reasons!
+		},
+	}
+
+	// registry
+	model.Registry = &models.VCHRegistry{
+		Insecure:               vchConfig.Registry.InsecureRegistries,
+		Whitelist:              vchConfig.Registry.RegistryWhitelist,
+		Blacklist:              vchConfig.Registry.RegistryBlacklist,
+		CertificateAuthorities: asPemCertificates(vchConfig.Certificate.RegistryCertificateAuthorities),
+		ImageFetchProxy:        asImageFetchProxy(vchConfig.ExecutorConfig.Sessions[config.VicAdminService], config.VICAdminHTTPProxy, config.VICAdminHTTPSProxy),
+	}
+
+	// runtime
+	model.Runtime = &models.VCHRuntime{}
+
+	installerVer := version.GetBuild()
+	upgradeStatus := upgradeStatusMessage(op, vch, installerVer, vchConfig.Version)
+	model.Runtime.UpgradeStatus = upgradeStatus
+
+	powerState, err := vch.PowerState(op)
+	if err != nil {
+		powerState = "error"
+	}
+	model.Runtime.PowerState = string(powerState)
+
+	if public := vchConfig.ExecutorConfig.Networks["public"]; public != nil {
```

Comment:
This is not correct.
See https://github.com/vmware/vic/blob/master/cmd/docker/main.go#L224 for the port choice logic.

The API is always served on the client network role (unless something is very wrong with current vic-machine) which MAY share an interface with the public network role.

Reply:
Filed #6728 for this.